File: solutions/fsharp/log-levels/LogLevels.fs
module LogLevels

open System.Text.RegularExpressions

let rxParseLogLine = Regex("\[(\w+?)\]:\s*(.+?)\s*$", RegexOptions.Compiled)

let message (logLine: string) : string =
    rxParseLogLine.Match(logLine).Groups[2].Value

let logLevel (logLine: string) : string =
    rxParseLogLine
        .Match(logLine)
        .Groups[ 1 ]
        .Value.ToLower()

let reformat (logLine: string) : string =
    $"{message logLine} ({logLevel logLine})"

File: solutions/julia/log-levels/log-levels.jl
function message(msg)
    parse(msg)[2]
end

function log_level(msg)
    lowercase(parse(msg)[1])
end

function reformat(msg)
    "$(message(msg)) ($(log_level(msg)))"
end

function parse(msg)
    match(r"\[(.*)\]:\s*(.*)", strip(msg)).captures
end